crates/forge/tests/it/revive/cheats_individual.rs

```diff
@@ -0,0 +1,177 @@
+    };
+}
+
+revive_cheat_test_original!(test_nonce, "Nonce");
+revive_cheat_test!(test_coinbase, "CoinBase");
+revive_cheat_test!(test_warp, "Warp");
+// TOFIX Fails  blockhash
+revive_cheat_test!(test_roll, "Roll");
+revive_cheat_test!(test_chainid, "ChainId");
+revive_cheat_test!(test_deal, "Deal");
+revive_cheat_test!(test_get_block_timestamp, "GetBlockTimestamp");
+revive_cheat_test!(test_get_block_number, "getBlockNumber");
+revive_cheat_test_original!(test_expect_emit, "ExpectEmit");
+// TOFIX
+revive_cheat_test_original!(test_expect_revert, "ExpectRevert");
+// TOFIX
+revive_cheat_test_original!(test_expect_call, "ExpectCall");
+// vm.fee() doesn't work correctly in revive mode
+// revive_cheat_test!(test_fee, "Fee");
+// vm.prevrandao() doesn't work correctly in revive mode
+// revive_cheat_test!(test_prevrandao, "Prevrandao");
+revive_cheat_test_original!(test_load, "Load");
+// Not implemented
+// revive_cheat_test_original!(test_access_list, "AccessList");
+revive_cheat_test_original!(test_addr, "Addr");
+// Not implemented vm.setArbitraryStorage
+// revive_cheat_test_original!(test_arbitrary_storage, "ArbitraryStorage");
+// SKIP it should not affect pallet-revive execution
+// revive_cheat_test_original!(test_assert, "Assert");
+// SKIP it should not affect pallet-revive execution
+// revive_cheat_test!(test_assume, "Assume");
+revive_cheat_test_original!(test_assume_no_revert, "AssumeNoRevert");
+// vm.attachBlob vm.broadcast does not work
+revive_cheat_test_original!(test_attach_blob, "AttachBlob");
+// vm.attachDelegation vm.broadcast does not work
+// revive_cheat_test_original!(test_attach_delegation, "AttachDelegation");
+// SKIP it should not affect pallet-revive execution
+// revive_cheat_test_original!(test_base64, "Base64");
+// Compilation error
+// revive_cheat_test!(test_blob_base_fee, "BlobBaseFee");
+revive_cheat_test_original!(test_blobhashes, "Blobhashes");
+// vm.broadcast does not work
+// revive_cheat_test_original!(test_broadcast, "Broadcast");
+//  vm.broadcastRawTransaction does not work
+// revive_cheat_test_original!(test_broadcast_raw_transaction, "BroadcastRawTransaction");
+// vm.cloneAccount not tested
+revive_cheat_test_original!(test_clone_account, "CloneAccount");
+// not supported in polkadot
+// revive_cheat_test_original!(test_cool, "Cool");
+// vm.copyStorage vm.setArbitraryStorage not implemented
+revive_cheat_test_original!(test_copy_storage, "CopyStorage");
+//  vm.deployCode not implemented
+revive_cheat_test_original!(test_deploy_code, "DeployCode");
+// SKIP it should not affect pallet-revive execution
+// revive_cheat_test_original!(test_derive, "Derive");
+// SKIP it should not affect pallet-revive execution
+// revive_cheat_test_original!(test_ens_namehash, "EnsNamehash");
+// SKIP it should not affect pallet-revive execution
+// revive_cheat_test_original!(test_env, "Env");
+// EXTCODECOPY compilation issue
+revive_cheat_test_original!(test_etch, "Etch");
```

Comment:
@smiasojed etch, prank, mock tests were already ported by me and passing here: https://github.com/paritytech/foundry-polkadot/blob/b6b6f0c3e958dcbecb244f7f8c692a70266d07c1/crates/forge/tests/it/revive/mod.rs#L5

Reply:
yes I saw it, I will clean this up. But anyway original etch is failing  - probably issue with EXTCODECOPY - but it should not affect our  EVM path.

Reply:
yeah EXTCODECOPY compilation was the reason I  modified. the test slightly.

Reply:
error out

Reply:
It means that if any project uses opcodes not supported in PVM, the EVM tests will fail. I’m not sure if this is the right approach.